src/ui/common/src/components/layouts/data_previewer.tsx

```diff
@@ -101,12 +107,45 @@ const DataPreviewer: React.FC<Props> = ({ previewData, error }) => {
           />
         </Box>
       );
-    } else {
+    } else if (
+      previewData.result.serialization_type === SerializationType.Image
+    ) {
+      const srcFromBase64 = 'data:image/png;base64,' + previewData.result.data;
+      data = <Image src={srcFromBase64} duration={0} fit="contain" />;
+    } else if (
+      previewData.result.serialization_type === SerializationType.Json
+    ) {
+      // Convert to pretty-printed version.
+      const prettyJson = JSON.stringify(
+        JSON.parse(previewData.result.data),
+        null,
+        2
+      );
+      data = (
+        <Typography sx={{ fontFamily: 'Monospace', whiteSpace: 'pre-wrap' }}>
```

Comment:
Might be a good idea to turn this Typography and the one down below on line 132 into their own components and pass in the data to be rendered there. From what I see, they pretty much are the same except for the content to be rendered.